### PR TITLE
Add test only genesis functions.

### PR DESCRIPTION
**Pull request by xjcaa**
This adds some test only functions that were made only accessible to friend modules and on genesis in this PR.

https://github.com/aptos-labs/aptos-core/pull/2842

These functions are useful for setting up genesis within an Aptos move test.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3235)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/reconfiguration.move [hunk 1]
```diff
@@ -150,4 +150,25 @@ module aptos_framework::reconfiguration {
             },
         );
     }
+
+    // For tests, skips the guid validation.
```
Comment (xjcaa) [inline aptos-move/framework/aptos-framework/sources/reconfiguration.move:154]: Wasn't sure how to get the GUID to equal 1 during testing so had to force create one for `aptos_framework`.

**Comment by xjcaa** [comment]
There may be others that would also be useful, but these were the ones I used specifically in my testing.

**Comment by xjcaa** [comment]
@movekevin tagging you here.

File: aptos-move/framework/aptos-framework/sources/stake.move [hunk 2]
```diff
@@ -2249,4 +2249,16 @@ module aptos_framework::stake {
         coin::merge(&mut stake_pool.pending_inactive, pending_inactive);
         stake_pool.locked_until_secs = locked_until_secs;
     }
+
+    #[test_only]
+    public fun initialize_for_test(
+        aptos_framework: &signer
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/stake.move:2255]: nit: This fits on the previous line

File: aptos-move/framework/aptos-framework/sources/reconfiguration.move [hunk 3]
```diff
@@ -150,4 +150,25 @@ module aptos_framework::reconfiguration {
...
+    #[test_only]
+    public fun initialize_for_test(
+        account: &signer
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/reconfiguration.move:157]: nit: This fits on the previous line

File: aptos-move/framework/aptos-framework/sources/reconfiguration.move (hunk 1 above)
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/reconfiguration.move:154]: This is fine 👍

**Review by movekevin — APPROVED** [review]
Thanks for making these changes 🙏

**Review by movekevin — COMMENTED** [review]
You need to regenerate a test. Can you cd into api in the repo and run 

env REGENERATE_GOLDENFILES=1 cargo test

**Comment by xjcaa** [comment]
> You need to regenerate a test. Can you cd into api in the repo and run
> 
> env REGENERATE_GOLDENFILES=1 cargo test

So I tried updating all my dependencies as per scripts/dev_setup.sh and merging with the latest main.

```
❯ env REGENERATE_GOLDENFILES=1 cargo test
    Finished test [unoptimized + debuginfo] target(s) in 1.79s
     Running unittests src/lib.rs (/Users/.../code/aptos-core/target/debug/deps/aptos_api-54b2b9187615e723)

running 112 tests
test runtime::tests::test_bootstrap_jsonprc_and_api_configured_at_different_port ... ignored
test tests::accounts_test::test_account_modules ... ignored
test tests::accounts_test::test_account_modules_structs ... ignored
test tests::accounts_test::test_account_resources_response ... ignored
test tests::accounts_test::test_get_account_modules_by_ledger_version ... ignored
test tests::accounts_test::test_get_module_aptos_config ... ignored
test tests::accounts_test::test_get_module_with_entry_functions ... ignored
test tests::accounts_test::test_get_core_account_data ... FAILED
test tests::accounts_test::test_get_account_resources_by_valid_account_address ... FAILED
test tests::accounts_test::test_get_account_resources_by_ledger_version ... FAILED
test tests::blocks_test::test_get_unknown_block_by_height ... FAILED
test tests::converter_test::test_value_conversion ... FAILED
test tests::accounts_test::test_get_account_resources_by_ledger_version_is_too_large ... FAILED
test tests::accounts_test::test_get_core_account_data_not_found ... FAILED
test tests::accounts_test::test_get_account_resources_by_invalid_address_missing_0x_prefix ... FAILED
test tests::accounts_test::test_get_account_resources_by_address_0x0 ... FAILED
test tests::blocks_test::test_get_genesis_block_by_height ... FAILED
test tests::events_test::test_get_events_by_struct_type_has_generic_type_parameter ... ignored
test tests::events_test::test_get_events_filter_by_limit_page_size ... ignored
test tests::accounts_test::test_get_account_resources_by_invalid_ledger_version ... ok
test tests::events_test::test_get_events_by_invalid_account_event_handle_struct_name ... FAILED
test tests::events_test::test_get_events_by_invalid_account_event_handle_struct_module ... FAILED
test tests::index_test::test_cors ... FAILED
test tests::events_test::test_get_events ... ok
test tests::events_test::test_get_events_by_invalid_account_event_handle_field_name ... ok
test tests::events_test::test_get_events_by_account_event_handle ... ok
test tests::index_test::test_return_bad_request_if_method_not_allowed ... ignored
test tests::index_test::test_returns_not_found_for_the_invalid_path ... ignored
test tests::events_test::test_get_events_by_invalid_key ... ok
test tests::events_test::test_get_events_filter_by_start_sequence_number ... ok
test tests::events_test::test_get_events_by_invalid_account_event_handle_struct_address ... ok
test tests::events_test::test_get_events_by_invalid_account_event_handle_field_type ... ok
test tests::invalid_post_request_test::test_invalid_entry_function_argument_data_type ... FAILED
test tests::invalid_post_request_test::test_invalid_entry_function_function_id ... FAILED
test tests::invalid_post_request_test::test_invalid_entry_function_function_name ... FAILED
test tests::invalid_post_request_test::test_invalid_entry_function_argument_address_type ... FAILED
test tests::invalid_post_request_test::test_invalid_entry_function_argument_u64_string ... FAILED
test tests::index_test::test_openapi_spec ... FAILED
test tests::invalid_post_request_test::test_invalid_sender_address ... ok
test tests::index_test::test_cors_forbidden ... FAILED
test tests::invalid_post_request_test::test_invalid_sequence_number ... ok
test tests::index_test::test_cors_on_non_200_responses ... ok
test tests::index_test::test_get_index ... ok
test tests::index_test::test_health_check ... ok
test tests::invalid_post_request_test::test_invalid_entry_function_argument_address_string ... ok
test tests::state_test::test_get_account_module ... FAILED
test tests::invalid_post_request_test::test_invalid_entry_function_module_name ... FAILED
test tests::state_test::test_get_account_module_by_invalid_address ... FAILED
test tests::invalid_post_request_test::test_invalid_payload_data_type ... FAILED
test tests::state_test::test_get_account_resource ... FAILED
test tests::invalid_post_request_test::test_invalid_entry_function_module_address ... FAILED
test tests::invalid_post_request_test::test_invalid_payload_type ... ok
test tests::invalid_post_request_test::test_invalid_type_argument_data_type ... ok
test tests::string_resource_test::test_renders_move_acsii_string_into_utf8_string ... ignored
test tests::invalid_post_request_test::test_missing_entry_function_arguments ... ok
test tests::state_test::test_get_account_module_not_found ... ok
test tests::state_test::test_get_table_item ... FAILED
test tests::state_test::test_get_account_resource_by_invalid_address ... FAILED
test tests::state_test::test_get_account_resource_with_version_too_large ... FAILED
test tests::transactions_test::test_create_signing_message_rejects_no_content_length_request ... ignored
test tests::transactions_test::test_create_signing_message_rejects_payload_too_large_json_body ... ignored
test tests::state_test::test_get_account_resource_with_version ... FAILED
fatal runtime error: Rust cannot catch foreign exceptions
error: test failed, to rerun pass '--lib'

Caused by:
  process didn't exit successfully: `/Users/.../code/aptos-core/target/debug/deps/aptos_api-54b2b9187615e723` (signal: 6, SIGABRT: process abort signal)
  ```

**Comment by davidiw** [comment]
I'll get this to land for you...

**Comment by movekevin** [comment]
I'll commandeer these changes in https://github.com/aptos-labs/aptos-core/pull/3268 and push them through the pipeline 🙏

**Comment by davidiw** [comment]
indeed... this is now merged.

**Comment by davidiw** [comment]
we'll land a pr shortly that wiill ... hopefully ... eliminate your friction next time.

thanks for the contribution.

**Pull request closed without merging**